torch/_tensor.py

```diff
@@ -227,6 +241,9 @@ def __reduce_ex__(self, proto):
         if has_torch_function_unary(self):
             return handle_torch_function(Tensor.__reduce_ex__, (self,), self, proto)
         func, args = self._reduce_ex_internal(proto)
+        # sizes / strides cache needs to be cleared here because it'll just be recached
+        # if cleared earlier. Note that state references the -actual- tensor dict.
+        _clear_cached_sizes_strides(self)
```

Comment:
Should we make this a method on the class that user can also override? Or we expect them to override `__reduce_ex__` to do cleanup before they call `super().__reduce_ex__(` ?

Reply:
Hm I like the idea of making this an overrideable class method. Is the name good or should it be generalized some to cover the general concept of clearing non-serializable cached stuff, wdyt?